app/src/lib/git/diff.ts

```diff
@@ -54,24 +109,49 @@ export function getWorkingDirectoryDiff(repository: Repository, file: WorkingDir
     //
     // citation in source:
     // https://github.com/git/git/blob/1f66975deb8402131fbf7c14330d0c7cdebaeaa2/diff-no-index.c#L300
-    opts = { successExitCodes: new Set([ 0, 1 ]) }
-    args = [ 'diff', '--no-ext-diff', '--no-index', '--patch-with-raw', '-z', '--no-color', '--', '/dev/null', file.path ]
+    successExitCodes = new Set([ 0, 1 ])
+
+    args = [ 'diff', '--no-ext-diff', '--no-index', '--patch-with-raw', '-z', '--no-color' ]
+
+    // enable --binary unless we know it's an image format
+    const extension = Path.extname(file.path)
+    if (!imageFileExtensions.has(extension)) {
+      args.push('--binary')
```

Comment:
This check makes me feel bad but if we enable this for known binary formats Git is actually eager to create a patch that can be applied for these binary files, so it no longer shows the "Binary files differ" message. Annoying, but perhaps there's some flag I've missed here.